File: array and string/pascals_triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        if numRows == 1:
            return [[1]]
        
        if numRows == 2:
            return [[1], [1,1]]
        
        else:
            l = [[1], [1,1]]
            
            for i in range(2,numRows):
                sub = []
                for j in range(len(l[len(l)-1])-1):
                    sum_ = l[len(l)-1][j] + l[len(l)-1][j+1]
                    sub.append(sum_)
                sub.insert(0,1)
                sub.append(1)
                l.append(sub)
        return l

File: array and string/pascals_triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        if (numRows == 0)
        {
            List<List<Integer>> a1 = new ArrayList<List<Integer>>();
            return a1;
        }
        if (numRows == 1)
        {
            List<List<Integer>> arr = new ArrayList<List<Integer>>();
            List<Integer> a1 = new ArrayList<>();
            a1.add(1);
            arr.add(a1);
            return arr;
        }
        if (numRows == 2){
            List<List<Integer>> arr = new ArrayList<List<Integer>>();
            List<Integer> a1 = new ArrayList<>();
            a1.add(1);
            arr.add(a1);
            List<Integer> a2 = new ArrayList<>(); 
            a2.add(1);
            a2.add(1);
            arr.add(a2);
            return arr;
        }
        
        else
        {
            List<List<Integer>> arr = new ArrayList<List<Integer>>();
            List<Integer> a1 = new ArrayList<>();
            a1.add(1);
            arr.add(a1);
            List<Integer> a2 = new ArrayList<>(); 
            a2.add(1);
            a2.add(1);
            arr.add(a2);
            System.out.println(arr);

            for(int i=2; i<numRows; i++)
            {
                List<Integer> a3 = new ArrayList<>();
                for(int j=0; j < arr.get(i-1).size()-1; j++)
                {
                    int sum = arr.get(i-1).get(j) + arr.get(i-1).get(j+1);
                    a3.add(sum);
                }
                a3.add(0,1);
                a3.add(1);
                arr.add(a3);
            }
            
            return arr;
        }
            
    }
}